File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    lp := 0
    rp := 1
    max := 0

    for rp < len(prices) {
        if prices[lp] < prices[rp] {
            profit := prices[rp] - prices[lp]
            if profit > max {
                max = profit
            }
        } else {
            lp = rp
        }
        rp++
    }

    return max
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let lp = 0;
  let rp = 1;
  let max = 0;

  while (rp < prices.length) {
    if (prices[rp] > prices[lp]) {
        const profit = prices[rp] - prices[lp];
        max = Math.max(profit, max);
    } else {
        lp = rp;
    }

    rp++;
  }  

  return max;
};